railties/lib/rails/templates/rails/mailers/email.html.erb

```diff
@@ -61,14 +61,14 @@
 <body>
 <header>
   <dl>
-    <% if @email.respond_to?(:smtp_envelope_from) && Array(@email.from) != Array(@email.smtp_envelope_from) %>
+    <% if @email.from != [@email.smtp_envelope_from] %>
```

Comment:
:+1: for removing `respond_to?(:smtp_envelope_from)`.  It was implemented in `mail` v2.5.4 by https://github.com/mikel/mail/commit/4875bc2ba1670289a34a155569af2f51abddc316, and Rails [requires `mail` >= 2.5.4](https://github.com/rails/rails/blob/a6099ed2b7a544d8fe67688e8a0f9237847404ee/actionmailer/actionmailer.gemspec#L41).

But I kind of prefer the usage of `Array()` here.  It makes it clear that we are definitely comparing the same type of value.

```suggestion
    <% if Array(@email.from) != Array(@email.smtp_envelope_from) %>
```